### DISCUSSION TITLE
Sample gRPC Dataportal?

### DISCUSSION THREAD

**Discussion by lacota** [MarimerLLC/csla#2428] [Questions]
Is there a sample gRPC dataportal? I saw Rocky's blog but am unclear about the client configuration. Also, does anyone have experience doing unit testing using WebApplicationFactory for the DAL?

**Comment by rockfordlhotka** [comment]
I don't think I have a sample that specifically uses the gRPC channel. The steps to use it are:

1. Add the `Csla.Channels.Grpc` package to your client and server projects
2. Configure the server endpoint
3. Configure the client to use the `GprcProxy` data portal proxy type
4. Enjoy

Only step 2 is a little unique, in that you use the `Csla.Channels.Grpc.GrpcPortal` type to create the endpoint. Microsoft has a [doc showing how this is done](https://docs.microsoft.com/en-us/aspnet/core/grpc/aspnetcore?view=aspnetcore-5.0&tabs=visual-studio). Just replace their `GreeterService` with the `GrpcPortal` type.
